bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -842,6 +853,69 @@ func (r *KubeadmConfigReconciler) resolveUsers(ctx context.Context, cfg *bootstr
 	return collected, nil
 }
 
+func (r *KubeadmConfigReconciler) resolveDiscoveryKubeConfig(config *bootstrapv1.KubeadmConfig) (*bootstrapv1.File, error) {
+	cfg := config.Spec.JoinConfiguration.Discovery.File
+	if cfg == nil || cfg.KubeConfig == nil {
+		return nil, errors.New("no discovery configuration file to resolve")
+	}
+	cluster := clientcmdv1.Cluster{
+		Server:                   cfg.KubeConfig.Cluster.Server,
+		TLSServerName:            cfg.KubeConfig.Cluster.TLSServerName,
+		InsecureSkipTLSVerify:    cfg.KubeConfig.Cluster.InsecureSkipTLSVerify,
+		CertificateAuthorityData: cfg.KubeConfig.Cluster.CertificateAuthorityData,
+		ProxyURL:                 cfg.KubeConfig.Cluster.ProxyURL,
+	}
+	user := clientcmdv1.AuthInfo{
+		ClientCertificateData: cfg.KubeConfig.User.ClientCertificateData,
+		ClientKeyData:         cfg.KubeConfig.User.ClientKeyData,
+		Token:                 cfg.KubeConfig.User.Token,
+	}
+	if cfg.KubeConfig.User.AuthProvider != nil {
+		user.AuthProvider = &clientcmdv1.AuthProviderConfig{
+			Name:   cfg.KubeConfig.User.AuthProvider.Name,
+			Config: cfg.KubeConfig.User.AuthProvider.Config,
+		}
+	}
+	if cfg.KubeConfig.User.Exec != nil {
+		user.Exec = &clientcmdv1.ExecConfig{
+			Command:            cfg.KubeConfig.User.Exec.Command,
+			Args:               cfg.KubeConfig.User.Exec.Args,
+			APIVersion:         cfg.KubeConfig.User.Exec.APIVersion,
+			InstallHint:        cfg.KubeConfig.User.Exec.InstallHint,
+			ProvideClusterInfo: cfg.KubeConfig.User.Exec.ProvideClusterInfo,
+			InteractiveMode:    clientcmdv1.ExecInteractiveMode(cfg.KubeConfig.User.Exec.InteractiveMode),
+		}
+		for _, env := range cfg.KubeConfig.User.Exec.Env {
+			user.Exec.Env = append(user.Exec.Env, clientcmdv1.ExecEnvVar{Name: env.Name, Value: env.Value})
+		}
+	}
+	kubeconfig := clientcmdv1.Config{
+		CurrentContext: "default",
+		Contexts: []clientcmdv1.NamedContext{
+			{
+				Name: "default",
+				Context: clientcmdv1.Context{
+					Cluster:  "default",
+					AuthInfo: "default",
+				},
+			},
+		},
+		Clusters:  []clientcmdv1.NamedCluster{{Name: "default", Cluster: cluster}},
+		AuthInfos: []clientcmdv1.NamedAuthInfo{{Name: "default", AuthInfo: user}},
+	}
```

Comment:
The context and other names can be anything, can also default it to the KubeadmConfig name if we prefer?

Reply:
To me default seems to be simpler. If someone actually wants to refer to the context they would have to calculate it otherwise, I think